src/audio/volume/volume_ipc4.c

```diff
@@ -297,6 +307,51 @@ static int volume_set_attenuation(struct processing_module *mod, const uint8_t *
 	return 0;
 }
 
+static int volume_set_switch(struct processing_module *mod, const uint8_t *data,
+			     int data_size)
+{
+	struct vol_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	struct sof_ipc4_control_msg_payload *ctl;
+	uint32_t i, channels_count;
```

Comment:
Just wondering the reason to change the type? Because uint32_t is widely used among the source code. Is there a plan to replace uint32_t in the future? Just curious.

Reply:
Unless you have a specific reason to use 32bits, you should let the compiler and architecture decide what size is optimal. This is especially true for local counters that have very limited interaction with the rest of the software.

Reply:
Also, `signed` allows the compiler to perform optimizations that the better defined `unsigned` does not:
https://blog.llvm.org/2011/05/what-every-c-programmer-should-know.html
https://lwn.net/Articles/959189/

This code does not seem to be in the critical path so it does not matter here. But those are things to always keep in mind. Many people think C gives simple, "direct" access to the hardware. It's much more complicated
https://queue.acm.org/detail.cfm?id=3212479 ("C Is Not a Low-level Language")
https://wiki.sei.cmu.edu/confluence/display/c/INT02-C.%2BUnderstand%2Binteger%2Bconversion%2Brules
etc.